pandas/core/groupby.py

```diff
@@ -1455,6 +1455,26 @@ def expanding(self, *args, **kwargs):
         from pandas.core.window import ExpandingGroupby
         return ExpandingGroupby(self, *args, **kwargs)
 
+    def _fill(self, how, limit=None):
+        labels, _, _ = self.grouper.group_info
+
+        # Need int value for Cython
+        if limit is None:
+            limit = -1
+        output = {}
+        if type(self) is DataFrameGroupBy:
+            for grp in self.grouper.groupings:
+                ser = grp.group_index.take(grp.labels)
+                output[ser.name] = ser.values
+        for name, obj in self._iterate_slices():
+            indexer = np.zeros_like(labels)
+            mask = isnull(obj.values).view(np.uint8)
+            libgroupby.group_fillna_indexer(indexer, mask, labels, how,
```

Comment:
maybe make a routine that shares code with how the shift_indexer is called (I am talking about in python space here)